src/Nextcloud/Provider.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace SocialiteProviders\Nextcloud;
+
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'NEXTCLOUD';
+
+    /**
+     * {@inheritdoc}
+     */
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase($this->getInstanceUri().'/index.php/apps/oauth2/authorize', $state);
```

Comment:
Looking at the docs, you shouldn't need the `index.php`?
https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/oauth2.html

Reply:
You are right, this is optional. Will fix it later, thanks.